bin/main.rb

```diff
@@ -1,2 +1,71 @@
 #!/usr/bin/env ruby
-puts 'Hello world!'
+
+puts 'WELCOME TO TIC-TAC-TOE GAME !!!'
+
+# create an empty game board
+@board = %w[1 2 3 4 5 6 7 8 9]
+
+# create game tokens as an array of 2 items
+@game_tokens = %w[X O]
+
+# create an empty hash to hold player and token info for each player
+# that is, player_one: 'X', player_two: 'O' or vice versa
+@assigned_tokens = {}
+
+# register two players
+puts 'Player 1, input name: '
+@player_one = gets.chomp
+
+puts 'Player 2, input name: '
+@player_two = gets.chomp
+
+# allow first player choose a game token X or O
+puts "#{player_one}, Choose a token: 'X' or 'O'"
+
+while selected_token == gets.chomp
```

Comment:
- [ ] The while loop should depend on a boolean variable to terminate and not from an input request change it to something like the following:

```ruby
game_on = true

# initial instructions for player

while game_on
  # loop for each move
  
  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```